### PR TITLE
Added reading of exc_mini_frequency and inh_mini_frequency

### PR DESCRIPTION
**Pull request by iomaganaris**
- In MVD3 and SONATA files
- Added the corresponding fields to circuit.mvd3 and nodes.h5 to
  avoid adding new file
- Added tests in C++ and Python

### CODE REVIEW

File: python/mvdtool.cpp [hunk 1]
```diff
@@ -281,6 +289,42 @@ PYBIND11_MODULE(mvdtool, mvd) {
...
+        .def("exc_mini_frequencies", [](const File& f, int offset) {
+                Range r(offset, 1);
+                return f.getExcMiniFrequencies(r)[0];
+             })
+        .def("exc_mini_frequencies", [](const File& f, int offset, int count) {
+                Range r(offset, count);
+                auto res = f.getExcMiniFrequencies(r);
+                return py::array(res.size(), res.data());
+             })
+        .def("exc_mini_frequencies", [](const File& f, pyarray<size_t> idx) {
```
Comment (tristan0x) [inline python/mvdtool.cpp]:
```suggestion
        .def("exc_mini_frequencies", [](const File& f, const pyarray<size_t>& idx) {
```

File: python/mvdtool.cpp [hunk 2]
```diff
@@ -281,6 +289,42 @@ PYBIND11_MODULE(mvdtool, mvd) {
...
+        .def("inh_mini_frequencies", [](const File& f, int offset) {
+                Range r(offset, 1);
+                return f.getInhMiniFrequencies(r)[0];
+             })
+        .def("inh_mini_frequencies", [](const File& f, int offset, int count) {
+                Range r(offset, count);
+                auto res = f.getInhMiniFrequencies(r);
+                return py::array(res.size(), res.data());
+             })
+        .def("inh_mini_frequencies", [](const File& f, pyarray<size_t> idx) {
```
Comment (tristan0x) [inline python/mvdtool.cpp]:
```suggestion
        .def("inh_mini_frequencies", [](const File& f, const pyarray<size_t>& idx) {
```

File: tests/unit/tests_mvd3.cpp [hunk 3]
```diff
@@ -322,6 +322,36 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
 
+BOOST_AUTO_TEST_CASE( basicExcFreq )
+{
+    using namespace MVD3;
+
+    MVD3File file(MVD3_FILENAME);
+
+    std::vector<double> exc_frequencies = file.getExcMiniFrequencies();
+
+    BOOST_CHECK_EQUAL(exc_frequencies[0], static_cast<double>(0.63));
```
Comment (tristan0x) [inline tests/unit/tests_mvd3.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(exc_frequencies[0], 0.63);
```

File: tests/unit/tests_mvd3.cpp [hunk 4]
```diff
@@ -322,6 +322,36 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(exc_frequencies[100], static_cast<double>(0.122));
```
Comment (tristan0x) [inline tests/unit/tests_mvd3.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(exc_frequencies[100], 0.122);
```

File: tests/unit/tests_mvd3.cpp [hunk 5]
```diff
@@ -322,6 +322,36 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(exc_frequencies[200], static_cast<double>(0.04));
```
Comment (tristan0x) [inline tests/unit/tests_mvd3.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(exc_frequencies[200], 0.04);
```

File: tests/unit/tests_mvd3.cpp [hunk 6]
```diff
@@ -322,6 +322,36 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+
+BOOST_AUTO_TEST_CASE( basicInhFreq )
+{
+    using namespace MVD3;
+
+    MVD3File file(MVD3_FILENAME);
+
+    std::vector<double> inh_frequencies = file.getInhMiniFrequencies();
+
+    BOOST_CHECK_EQUAL(inh_frequencies[0], static_cast<double>(0.012));
```
Comment (tristan0x) [inline tests/unit/tests_mvd3.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(inh_frequencies[0], 0.012);
```

File: tests/unit/tests_mvd3.cpp [hunk 7]
```diff
@@ -322,6 +322,36 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(inh_frequencies[100], static_cast<double>(0.013));
```
Comment (tristan0x) [inline tests/unit/tests_mvd3.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(inh_frequencies[100], 0.013);
```

File: tests/unit/tests_mvd3.cpp [hunk 8]
```diff
@@ -322,6 +322,36 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(inh_frequencies[200], static_cast<double>(0.014));
```
Comment (tristan0x) [inline tests/unit/tests_mvd3.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(inh_frequencies[200], 0.014);
```

File: tests/unit/tests_sonata.cpp [hunk 9]
```diff
@@ -251,6 +251,32 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
 }
 
 
+BOOST_AUTO_TEST_CASE( basicExcFreq )
+{
+    SonataFile file(SONATA_FILENAME_NEW_FORMAT);
+
+    std::vector<double> exc_frequencies = file.getExcMiniFrequencies();
+
+    BOOST_CHECK_EQUAL(exc_frequencies[0], static_cast<double>(0.63));
```
Comment (tristan0x) [inline tests/unit/tests_sonata.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(exc_frequencies[0], 0.63);
```

File: tests/unit/tests_sonata.cpp [hunk 10]
```diff
@@ -251,6 +251,32 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(exc_frequencies[100], static_cast<double>(0.122));
```
Comment (tristan0x) [inline tests/unit/tests_sonata.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(exc_frequencies[100], 0.122);
```

File: tests/unit/tests_sonata.cpp [hunk 11]
```diff
@@ -251,6 +251,32 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(exc_frequencies[200], static_cast<double>(0.04));
```
Comment (tristan0x) [inline tests/unit/tests_sonata.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(exc_frequencies[200], 0.04);
```

File: tests/unit/tests_sonata.cpp [hunk 12]
```diff
@@ -251,6 +251,32 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+}
+
+
+BOOST_AUTO_TEST_CASE( basicInhFreq )
+{
+    SonataFile file(SONATA_FILENAME_NEW_FORMAT);
+
+    std::vector<double> inh_frequencies = file.getInhMiniFrequencies();
+
+    BOOST_CHECK_EQUAL(inh_frequencies[0], static_cast<double>(0.012));
```
Comment (tristan0x) [inline tests/unit/tests_sonata.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(inh_frequencies[0], 0.012);
```

File: tests/unit/tests_sonata.cpp [hunk 13]
```diff
@@ -251,6 +251,32 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(inh_frequencies[100], static_cast<double>(0.013));
```
Comment (tristan0x) [inline tests/unit/tests_sonata.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(inh_frequencies[100], 0.013);
```

File: tests/unit/tests_sonata.cpp [hunk 14]
```diff
@@ -251,6 +251,32 @@ BOOST_AUTO_TEST_CASE( basicTestSynClass )
...
+    BOOST_CHECK_EQUAL(inh_frequencies[200], static_cast<double>(0.014));
```
Comment (tristan0x) [inline tests/unit/tests_sonata.cpp]:
```suggestion
    BOOST_CHECK_EQUAL(inh_frequencies[200], 0.014);
```

**Comment by iomaganaris** [comment]
> `0.63` is a double by default.
> See https://en.cppreference.com/w/cpp/language/floating_literal

Good point! Thank you @tristan0x . I have removed the cast to `double` from all the places that existed in the `boost` tests.

File: python/mvdtool.cpp (hunk 1 above)
Comment (iomaganaris) [reply python/mvdtool.cpp]: Also good point :+1:

**Pull request merged**